core/scripts.py

```diff
@@ -3581,6 +3584,61 @@ def _update_list_with_new_entries(self, custom_list: CustomList):
         custom_list.auto_update_last_update = datetime.datetime.now()
 
 
+class LoanNotificationsScript(Script):
+    """Notifications must be sent to Patrons based on when their current loans
+    are expiring"""
+
+    # Days before on which to send out a notification
+    LOAN_EXPIRATION_DAYS = [5, 1]
+    BATCH_SIZE = 100
+
+    def do_run(self):
+        self.log.info("Loan Notifications Job started")
+        _query = self._db.query(Loan).order_by(Loan.id)
+        last_loan_id = None
+        processed_loans = 0
+        for _ in range(100000):
```

Comment:
Yyou can call him "Courage, the Cowardly while true".
I've added a comment to make it clear that this is a fail-safe max number of loops.

Reply:
Can you say a little more in the comment about why you think we need a failsafe here (e.g., result set shifting with each batch) and possibly add a TODO to bring us back here to fix in the future.

Reply:
There is absolutely no reason I have to believe this will actually become an infinite loop. It's just some defensive code, it can become a `while true` now if required. Should I change it?

Reply:
My concern about the defensive coding here is that if it actually came into play, it would silently mask the underlying problem. So, if we keep it, it would be good to add an `else` clause to the `for` loop to emit an error about what just happened. We might also want to pick a number for the `range()` call this is tied more closely to our state (e.g., total-loan-count/batch-size). Or we could just go with `while True` and cross our fingers. 😄 🤞🏽

Reply:
From my perspective I wouldn't call this a blocker, but I thought I'd throw my 2 cents 🪙 in here, on my preferences for loops. 

Personally I prefer `while True` over `for _ in range(100000)`. But what I'd most prefer is something with a more explicit conditional like `while len(loans) != 0`, but that would take some restructuring of the logic here.

Reply:
Updated this to be a `while true` since we already do a `len == 0: break` in the loop